lib/geordi/db_cleaner.rb

```diff
@@ -0,0 +1,183 @@
+          mysql_command = 'mysql -uroot'
+          mysql_command << " #{extra_flags}" unless extra_flags.nil?
+          unless File.exist? File.join(Dir.home, 'my.cnf')
+            warn 'You do not have $HOME/.my.cnf. You will have to enter your MySQL root password a lot!'
+            mysql_command << ' -p'
+          end
+          Open3.popen3("#{mysql_command} -e 'QUIT'") do |stdin2, stdout2, stderr2, thread2|
+            fail 'Could not connect to MySQL/MariaDB' unless thread2.value.exitstatus == 0
+          end
+        elsif mysql_error == '2013'
+          fail 'MySQL/MariaDB connection failed, is this the correct port?'
+        end
+      end
+      database_list = list_all_mysql_dbs(mysql_command)
+      # confirm_deletion includes option for whitelist editing
+      deletable_dbs = confirm_deletion('mysql', database_list)
+      return if deletable_dbs.nil?
+      deletable_dbs.each do |db|
+        `#{mysql_command} -e 'DROP DATABASE \`#{db}\`;'`
+      end
+    end
+
+    def clean_postgres(extra_flags)
+      announce 'Dropping PostgreSQL databases'
+      postgres_command = 'sudo -u postgres psql'
+      unless extra_flags.nil?
+        begin
+          port = Integer(extra_flags.split('=')[1])
+          fail "Port #{port} is not open" unless is_port_open? port
+        rescue ArgumentError
+          socket = extra_flags.split('=')[1]
+          fail "Socket #{socket} does not exist" unless File.exist? socket
+        end
+        postgres_command << " #{extra_flags}"
+      end
+      Open3.popen3("#{postgres_command} -c ';'") do |stdin, stdout, stderr, thread|
+        fail 'Could not connect to PostgreSQL' unless thread.value.exitstatus == 0
+      end
+      database_list = `#{postgres_command} -t -A -c 'SELECT DATNAME FROM pg_database WHERE datistemplate = false;'`.split
+      deletable_dbs = confirm_deletion('postgres', database_list)
+      return if deletable_dbs.nil?
+      deletable_dbs.each do |db|
+        note "Dropping database `#{db}`."
+        `#{postgres_command} -c 'DROP DATABASE "#{db}";'`
+      end
+    end
+
+    def list_all_mysql_dbs(mysql_command)
+      `#{mysql_command} -B -N -e 'show databases'`.split
+    end
+    private :list_all_mysql_dbs
+
+    def whitelist_fname(dbtype)
+      File.join(@whitelist_directory, dbtype) << '.yaml'
+    end
+
+    def confirm_deletion(dbtype, database_list)
+      proceed = ''
+      until %w[y n].include? proceed
+        deletable_dbs = filter_whitelisted(dbtype, database_list)
+        if deletable_dbs.empty?
+          note "No #{dbtype} databases found that were not whitelisted"
+          if prompt('Edit the whitelist? [y]es or [n]o') == 'y'
+            proceed = 'e'
+          else
+            return []
+          end
+        end
+        if proceed.empty?
+          note "The following #{dbtype} databases are not whitelisted and will be deleted:"
+          deletable_dbs.sort.each do |db|
+            note db
```

Comment:
Please simply `puts` the database, as `note` is intended for comments and information, not output.